code/modules/mob/interactive.dm

```diff
@@ -726,6 +723,8 @@
 					TARGET = pick(target_filter(favouredObjIn(urange(MAX_RANGE_FIND,src,1))))
 				if(46 to 50)
 					TARGET = pick(target_filter(oview(MIN_RANGE_FIND,src)))
+		else if(isTraitor && traitorTarget)
+			TARGET = traitorTarget
```

Comment:
Pathing is technically infinite and I haven't seen any problems with this. If it does bug out in further testing on the server then I'll fix it.